be/src/exprs/vectorized/string_functions.cpp

```diff
@@ -968,8 +987,14 @@ static inline ColumnPtr repeat_not_const(const Columns& columns) {
 
         auto s = str_viewer.value(i);
         int32_t n = times_viewer.value(i);
-        n = std::min(n, (int32_t)(OLAP_STRING_MAX_LENGTH / s.size));
-        n = std::max(n, 1);
+
+        if (s.size > LIMIT / n || s.size * n > OLAP_STRING_MAX_LENGTH) {
```

Comment:
I think `s.size > LIMIT / n` can infer to `s.size * n > OLAP_STRING_MAX_LENGTH`. 
Why not only check `s.size * n > OLAP_STRING_MAX_LENGTH`?
And `LIMIT / n` will introduce divdie operation. I think this can be optimized to one divide operation?

Reply:
s.size * n may cause a overflow